tests/test_bulk_upload.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+import os
+from instructor import Instructor
+from data import is_valid_student_id, is_valid_grade
+
+def test_valid_student_id():
```

Comment:
If you want to use `is_valid_student_id` and `is_valid_grade`, please make sure in the program, whenever a grade or a student id is entered, they are checked with the same function.

Also, I think most tests/code are based on the assumption that grades are a floating-point data type. Please double-check that.